ocular/src/chain/client/query.rs

```diff
@@ -20,31 +20,43 @@ pub type SlashingQueryClient = slashing::query_client::QueryClient<Channel>;
 pub type StakingQueryClient = staking::query_client::QueryClient<Channel>;
 
 impl ChainClient {
-    pub fn check_for_grpc_address(&self) -> Result<(), GrpcError> {
-        if self.config.grpc_address.is_empty() {
-            return Err(GrpcError::MissingEndpoint(format!(
-                "no grpc address available for chain {}",
-                self.config.chain_id
-            )));
+    // Auth queries
+    pub async fn get_auth_query_client(&mut self) -> Result<AuthQueryClient, ChainClientError> {
+        let mut result: Result<AuthQueryClient, ChainClientError> =
+            Err(TxError::BroadcastError(String::from("Client connection never attempted.")).into());
+
+        // Get grpc address randomly each time; shuffles on failures
+        for _i in 0u8..self.connection_retry_attempts + 1 {
+            // Get an endpoint
+            let endpoint = match self.get_random_grpc_endpoint().await {
+                Ok(endpt) => endpt,
+                Err(err) => return Err(GrpcError::MissingEndpoint(err.to_string()).into()),
+            };
```

Comment:
we should default to an endpoint that has worked in the past instead of doing this every time by default. this could also be solved by caching the healthy client, but it's probably cleaner to just have a "default" endpoint to try once a healthy one is found. It can be shared by all clients.

Reply:
Sure we can have a default, do you think it would be nicer to set it in the cache, or have it in the client config somewhere?

Reply:
Seems appropriate for it to be config level (there was a field `grpc_endpoint` already). To be clear, by "default" I just mean "last healthy endpoint used". So like if we get an auth query client, it sets the endpoint field, and then later if we make a bank query client, it uses that same endpoint.

Reply:
Updated to bring back grpc_endpoint field and make it configurable and update to the last healthy endpoint connected to.